src/registerListeners.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionContext, TextDocument, workspace } from 'vscode';
+import { ext } from './extensionVariables';
+
+let lastUploadTime: number = 0;
+const hourInMilliseconds = 1000 * 60 * 60;
+
+export function registerListeners(ctx: ExtensionContext): void {
+    ctx.subscriptions.push(workspace.onDidSaveTextDocument(onDidSaveTextDocument));
+}
+
+function onDidSaveTextDocument(doc: TextDocument): void {
+    // If it's not a Dockerfile, or last upload time is within an hour, skip
+    if (doc.languageId !== 'dockerfile' || lastUploadTime + hourInMilliseconds > Date.now()) {
+        return;
+    }
+
+    lastUploadTime = Date.now();
+    ext.reporter.sendTelemetryEvent('dockerfilesave', { "lineCount": doc.lineCount.toString() }, {});

```

Comment:
I guess the Docker extension team is curious as to how big people's Dockerfiles are? 🤔

Reply:
Mostly, we want to see how much people are using the extension just to get the Dockerfile intellisense; because it seems to be the majority.

Reply:
For example, in a month we had ~300K machines activate the extension, but ~60K that did anything else. There's more ways to trigger activation than just opening a Dockerfile but that is probably the most common one...

Reply:
Interesting. I'd be curious as to how useful people actually find the editor features and what they use the most of. I imagine most people don't actually **need** code completion given that Dockerfiles aren't complicated beasts.